docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,289 @@
+
+To create, delete, and update seeds effectively in a declarative way and allow auto-scaling, a "creatable seed" resource along with a "set" and a "deployment"  of such creatable seeds should be introduced, similar to Kubernetes `Pod`, `ReplicaSet`, and `Deployment` (or to MCM `Machine`, `MachineSet`, and `MachineDeployment`) resources. With such resources (and their respective controllers), creating a new seed based on a template would become as simple as increasing the `replicas` field in the "set" resource, and updates to seeds could be performed in a rolling manner via the "deployment" resource. 
+
+In [Issue #2181](https://github.com/gardener/gardener/issues/2181) it is already proposed that the `use-as-seed` annotation is replaced by a dedicated `ShootedSeed` resource. The solution proposed here further elaborates on this idea.
+
+### ShootedSeeds
+
+The `ShootedSeed` resource is a dedicated custom resource that represents a "shooted seed" and properly replaces the `use-as-seed` annotation. This resource contains:
+
+* A Shoot template that contains the Shoot spec and parts of the metadata, such as labels and annotations.
+* A Seed template that contains the Seed spec and parts of the metadata, such as labels and annotations.
+* A `gardenlet` section that contains:
+    * Whether `gardenlet` is enabled or not.
+    * Certain aspects of the `gardenlet` deployment configuration, such as the number of replicas, the image, which bootstrap mechanism to use (bootstrap token / service account), etc.
+    * The `GardenletConfiguration` resource that contains controllers configuration, feature gates, etc.
+    
+A ShootedSeed allows fine-tuning the seed and the `gardenlet` configuration of shooted seeds in order to deviate from the global defaults, e.g. lower the concurrent sync for some of the seed's controllers or enable a feature gate only on certain seeds. Also, it could simplify the deletion protection of shooted seeds. Last but not least, ShootedSeeds could be used as the basis for creating and deleting seeds automatically via the `ShootedSeedSet` and `ShootedSeedDeployment` resources that are described in more details below.
+
+Unlike the `Seed` resource, the `ShootedSeed` resource is namespaced. If created in the `garden` namespace, the resulting seeds are globally available. If created in a project namespace, the resulting seeds automatically receive project specific taints and labels, and so can be used as "private seeds" by shoots in the project. The concept of private seeds / cloudprofiles is described in [Issue #2874](https://github.com/gardener/gardener/issues/2874).
```

Comment:
I haven't really dived much into private seeds, in this case I could also remove the mention of private seeds and regard `ShootedSeed` as a cluster resource.

Reply:
I think this boils down to what was discussed in the meeting last week, i.e., that the GEP is not focusing on one specific enhancement of Gardener, but rather on the foreseeable complete set of goals/topics that relate to the "automated seed management" topic. It's fine and welcome to have such document that covers the topic as a whole, but then it should indeed only mention the various aspects from a high-level. Currently, it's kind of mixed, some topics already contain implementation details, others don't, others contain a vague set of details (like this one about private seeds), but when asking back the real details are still missing.

Concrete proposal:
Keep this document as an overview (which topics belong to "automated seed management", what are they/which aspects are covered by them, what do they depend on (e.g., GEP7), and in which order they should be implemented). Once approved, you can still cut out the implementation details of the individual smaller topics and put them into a dedicated GEP that focuses on the realization of one specific topic and indeed discusses the various implementation details.

WDYT?

Reply:
> Keep this document as an overview (which topics belong to "automated seed management", what are they/which aspects are covered by them, what do they depend on (e.g., GEP7), and in which order they should be implemented)

This is more or less what I am already trying to do, so I fully agree. Most of the sections are missing implementation details exactly because I don't think it's very effective to go into such details at this stage.

> Once approved, you can still cut out the implementation details of the individual smaller topics and put them into a dedicated GEP that focuses on the realization of one specific topic and indeed discusses the various implementation details.

Again, I fully agree, except perhaps that if a change is fairly small I would put it into an issue rather than write a new GEP. This could be decided on a case by case basis.

Reply:
After a discussion with @rfranzke we agreed to keep the GEP as it is now. After removing the topics that are too future-looking (custom resource plugins, ShootedSeedDeployments), the GEP is sufficiently both coherent and relatively small in scope, so that further breaking down into multiple GEPs is not required. Note that a number of topics have already being moved to GEP-14 (see #3067).